fairscale/optim/oss.py

```diff
@@ -146,9 +146,22 @@ def state_dict(self) -> Dict[str, Any]:
             len(self._all_states) > 0
         ), "The optimizer state is not materialized, please call consolidate_state_dict on every replica beforehand"
 
+        # Flatten the state_dict, save the partition
```

Comment:
The code doesn't seem to flatten the state dict. It just converts the list to a dictionary. Is this comment correct?

Reply:
the flattening starts l153, I can move the comment down

Reply:
My concern is more that the comment does not seem to be totally consistent with the code. We don't really flatten state_dict, we just flatten the param_groups part of the state_dict. state itself is not flattened. Instead we convert the state field to a dictionary with integer keys. Why not flatten that also? If it can't be flattened, please add a comment as to why.

Reply:
ah ok, agreed, I thought it was more about the localization, I can definitely change the comment. I'm not sure of the benefit of flattening the state else, I don't know of any use case, so it was more a matter of not doing something not needed, but I can do the same for sure